content/docs/rsapi/Implement.md

```diff
@@ -0,0 +1,245 @@
+---
+title: 'Implement ReactiveSearch API'
+meta_title: 'ReactiveSearch API - Implementation guide'
+meta_description: 'ReactiveSearch API - Implementation guide. A offers a declarative interface to querying ElasticSearch, prevents query injections and lets you extend the API via Query Rules and Functions. '
+keywords:
+    - concepts
+    - appbase
+    - elasticsearch
+    - reactivesearch
+sidebar: 'docs'
+---
+
+## Concepts
+
+The `ReactiveSearch` API is derived from the `ReactiveSearch` library. If you're already using `ReactiveSearch` in your projects, then you will feel right at home because API maintains a 100% prop parity with the library. Even if you haven't worked with `ReactiveSearch` before, you should be able to follow along as we'll cover all aspects of `ReactiveSearch API` in this document.
+
+### Query Identifier(ID)
+
+Every query made via the `RS API` must have an `id` property defined. `id` is a unique identifier for each query. The `id` property is also useful to access the response for a particular query if multiple queries are defined within a single ReactiveSearch API call. The id can be referenced in the `react` property of any other queries that wish to also apply this query. We'll talk more about the usage of `react` property later and will explain how to reference an `id` in `react` property to combine multiple queries.
+
+### Type of Queries
+
+`ReactiveSearch API` has four types of queries, each serving a different use-case. You can decide that which query has to be used by defining the `type` property in the query object.
+
+#### Search (default)
+
+Search queries allow you to find documents for which the `value` matches in any of the fields specified in its `dataField` property. Search queries can take into account the text analysis process, partial words, as-you-type queries, can handle typo tolerance, offer response highlighting and search on synonyms. All of these are configurable by the search query's properties.
```

Comment:
@bietkul Can you use this section as a reference for how to write the other query type sections? (We want to cover the kinds of queries from a use-case perspective)